tempesta_fw/http_parser.c

```diff
@@ -1983,20 +1974,6 @@ __req_parse_accept(TfwHttpReq *req, unsigned char *data, size_t len)
 		__FSM_I_JMP(Req_I_Subtype);
 	}
 
-	__FSM_STATE(Req_I_AfterStar) {
-		if (c == '/')
-			__FSM_I_MOVE(Req_I_StarSlashStar);
-		if (IS_TOKEN(c))
-			__FSM_I_JMP(Req_I_Type);
-		return CSTR_NEQ;
-	}
-
-	__FSM_STATE(Req_I_StarSlashStar) {
-		if (c == '*')
-			__FSM_I_MOVE(Req_I_AcceptHtml);
-		return CSTR_NEQ;
-	}
-
 	__FSM_STATE(Req_I_AcceptHtml) {
```

Comment:
It seems `*` processing in
```C
        __FSM_STATE(Req_I_Slash) {
                if (IS_TOKEN(c))
                        __FSM_I_JMP(Req_I_Subtype);
                if (c == '*')
                        __FSM_I_MOVE(I_EoT);
```
doesn't make any sense becuase `*` is always treated as token. The same for `__h2_req_parse_accept()`.

Reply:
Thank you for pointing this out. I reviewed the parser function an made it a bit strict:now it doesn't allow asterisk character at the beginning of the `type` or `subtype`. Related tests was updated and documented.